test/index.spec.ts

```diff
@@ -0,0 +1,236 @@
+import { expect } from 'chai';
+import type { Interceptable } from 'undici';
+import { MockAgent, setGlobalDispatcher } from 'undici';
+import * as net from 'net';
+import * as fs from 'fs';
+import * as netCheck from '../lib/index';
+
+describe('network-check', () => {
+	const HOST = 'https://balena.io';
+	let mockAgent: MockAgent;
+	let mockPool: Interceptable;
+	before(() => {
+		mockAgent = new MockAgent();
+		setGlobalDispatcher(mockAgent);
+	});
+
+	describe('monitor', () => {
+		beforeEach(() => {
+			mockPool = mockAgent.get(HOST);
+			mockPool
+				.intercept({
+					path: '/test1',
+				})
+				.reply(200)
+				.persist();
+		});
+
+		afterEach(async () => {
+			await mockPool.close();
+		});
+
+		it('throws an error if checkFn is not a function', () => {
+			expect(() => {
+				netCheck.monitor(null as any, { url: HOST }, () => {
```

Comment:
I prefer to use
```suggestion
				// @ts-expect-error intentionally testing that invalid args fail
				netCheck.monitor(null, { url: HOST }, () => {
```
since that way it double up as a test that the invalid code also fails the type check